### ISSUE TITLE
Server started with delve stops immediately

### ISSUE THREAD

**Issue by Minivera** [go-delve/delve#1461]
# 1. What version of Delve are you using (`dlv version`)?
```
Delve Debugger
Version: 1.1.0
Build: $Id: 1990ba12450cab9425a2ae62e6ab988725023d5c $
```

# 2. What version of Go are you using? (`go version`)?
`go version go1.11.4 darwin/amd64`

# 3. What operating system and processor architecture are you using?
OSX, amd64

# 4. What did you do?
I am trying to add delve into one of our microservices containers when running locally. The application is prebuilt outside the container itself with `-gcflags "all=-N -l"`and is simply copied. I use a slightly modified version of the suggested dockerfile by JetBrains.

```
# Compile stage
FROM golang:1.11.4-alpine3.8 AS build-env
ENV CGO_ENABLED 0

# Compile Delve
RUN apk add --no-cache git
RUN go get github.com/derekparker/delve/cmd/dlv

FROM alpine:3.7

ENV PORT=8080
ENV DELVE_PORT=40000
ENV HEALTH_CHECK_PORT=8080
ENV HOST=0.0.0.0
ENV ENV=dev

EXPOSE $PORT $DELVE_PORT

# Allow delve to run on Alpine based containers.
RUN apk add --no-cache libc6-compat
RUN apk --no-cache add ca-certificates

COPY --from=build-env /go/bin/dlv /

ARG BIN
COPY $BIN ./api

ENTRYPOINT ["/dlv", "--listen=0.0.0.0:40000", "--headless=true", "--api-version=2", "--log=true", "exec", "./api"]
```

I then execute the container using a docker compose where I set everything I should be setting for delve. 

```yaml
...
    security_opt:
      - apparmor:unconfined
      - seccomp:unconfined
    cap_add:
      - SYS_PTRACE
    ports:
      - "40000:40000"
    expose:
      - "40000"
...
```

The server is a swagger server and it listens for the port `8080`.

# 5. What did you expect to see?
My application being alive and displayed when running `docker ps`. Or at least, a log of what it said before dying.

# 6. What did you see instead?
When I check the docker logs, all I see is this:

```
API server listening at: [::]:40000
```

And the container is already dead, but without any indication of what happened.

**Comment by derekparker** [comment]
No, that definitely does not seem right. The headless dlv server should continue running waiting for a connection. IIRC you're using an alpine container, can you try and get a shell within the container and run the command yourself and see if you get any more error output?

**Comment by Minivera** [comment]
I will do that tomorrow and let you know! Thanks

**Comment by dlsniper** [comment]
Have you built the application using alpine as well? If not then build the application using the same container type, as shown in the tutorial you referenced. It's likely that there's a problem between different environments which causes this.

**Comment by Minivera** [comment]
I will try that, our application is somewhat gigantic, so we build outside the containers themselves for simplicity sake.

**Comment by aarzilli** [comment]
Are you passing -w to ldflafgs or stripping the executable.

**Comment by Minivera** [comment]
After building in the container, the error changed. It's now the same error I had when trying the same thing on local with OSX. I was under the impression this was fixed, is it the same issue as #1138 ?

```
Could not create config directory: user: Current not implemented on linux/amd64.API server listening at: [::]:40000
INFO[0000] launching process with args: [./api]          layer=debugger
could not launch process: decoding dwarf section info at offset 0x0: too short
```

> Are you passing -w to ldflafgs or stripping the executable?

We're passing `-w` yes, should I remove that?

**Comment by aarzilli** [comment]
> We're passing -w yes, should I remove that?

Yes:

```
$ go tool link -h
usage: link [options] main.o
...
  -w	disable DWARF generation
```
